aws/resource_aws_wafregional_rule_test.go

```diff
@@ -0,0 +1,244 @@
+	var v waf.Rule
+	wafRuleName := fmt.Sprintf("wafrule%s", acctest.RandString(5))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafRegionalRuleDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: testAccAWSWafRegionalRuleConfig(wafRuleName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalRuleExists("aws_wafregional_rule.wafrule", &v),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_rule.wafrule", "name", wafRuleName),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_rule.wafrule", "predicates.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_rule.wafrule", "metric_name", wafRuleName),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSWafRegionalRule_changeNameForceNew(t *testing.T) {
+	var before, after waf.Rule
+	wafRuleName := fmt.Sprintf("wafrule%s", acctest.RandString(5))
+	wafRuleNewName := fmt.Sprintf("wafrulenew%s", acctest.RandString(5))
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafRegionalIPSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSWafRegionalRuleConfig(wafRuleName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalRuleExists("aws_wafregional_rule.wafrule", &before),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_rule.wafrule", "name", wafRuleName),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_rule.wafrule", "predicates.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_rule.wafrule", "metric_name", wafRuleName),
+				),
+			},
+			{
+				Config: testAccAWSWafRegionalRuleConfigChangeName(wafRuleNewName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalRuleExists("aws_wafregional_rule.wafrule", &after),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_rule.wafrule", "name", wafRuleNewName),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_rule.wafrule", "predicates.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_wafregional_rule.wafrule", "metric_name", wafRuleNewName),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSWafRegionalRule_disappears(t *testing.T) {
+	var v waf.Rule
+	wafRuleName := fmt.Sprintf("wafrule%s", acctest.RandString(5))
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafRegionalRuleDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSWafRegionalRuleConfig(wafRuleName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafRegionalRuleExists("aws_wafregional_rule.wafrule", &v),
+					testAccCheckAWSWafRegionalRuleDisappears(&v),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
```

Comment:
Because there has been a bug in the past affecting all WAF resources I'd like to see 1 more test here, specifically one changing tuples, see https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_waf_xss_match_set_test.go#L111-L174 for inspiration